### PR TITLE
[WIP] Migrate to new pipe APIs 

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -10,7 +10,6 @@
     <PublicSign Condition="'$(OS)' != 'Windows_NT'">true</PublicSign>
     <VersionSuffix Condition="'$(VersionSuffix)'!='' AND '$(BuildNumber)' != ''">$(VersionSuffix)-$(BuildNumber)</VersionSuffix>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-    <LangVersion>latest</LangVersion>
```
Comment (halter73) [inline Directory.Build.props]: reminder to revert

Comment (muratg) [reply Directory.Build.props]: Double reminder

File: NuGet.config [hunk 2]
```diff
@@ -5,5 +5,7 @@
     <add key="AspNetCore" value="https://dotnet.myget.org/F/aspnetcore-ci-dev/api/v3/index.json" />
     <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
     <add key="roslyn" value="https://dotnet.myget.org/F/roslyn/api/v3/index.json" />
+    <add key="corefxlab" value="https://dotnet.myget.org/F/dotnet-corefxlab/api/v3/index.json" />
+    <add key="corefx" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
```
Comment (muratg) [inline NuGet.config]: These to be removed as well, I assume... (once the bits are consumed?)

File: src/Kestrel.Transport.Sockets/Internal/SocketConnection.cs [hunk 3]
```diff
@@ -240,5 +243,16 @@ internal SocketConnection(Socket socket, PipeFactory pipeFactory, ISocketsTrace
                 _socket.Shutdown(SocketShutdown.Both);
             }
         }
+
+        private static ArraySegment<byte> GetArraySegment(Memory<byte> buffer)
```
Comment (halter73) [inline src/Kestrel.Transport.Sockets/Internal/SocketConnection.cs]: Is this called still?

**Pull request merged**